backend/controllers/announcements.js

```diff
@@ -0,0 +1,70 @@
+const Announcement = require("../models/Announcement");
+
+// @desc:    Get all announcements
+// @route:   GET /api/v1/announcements
+// @access:  Public
+exports.getAnnouncements = async (req, res, next) => {
+  let query;
+  console.log(req);
+  
+  if (req.params.campgroundId) {
+    console.log(req.params.campgroundId);
+    query = Announcement.find({
+      campground: req.params.campgroundId,
+    }).populate({
+      path: "campground",
+      select: "name",
+    });
+  }
+  else {
+    query = Announcement.find().populate({
+      path: "campground",
+      select: "name",
+    });
+  }
+
+  query = query.sort("startDate");
+
+  try {
+    const announcements = await query;
+
+    res.status(200).json({
+      success: true,
+      count: announcements.length,
+      data: announcements,
+    });
+    console.log("success", announcements.length);
+  } catch (err) {
+    console.log(err.stack);
+    res.status(400).json({
```

Comment:
This line and line 65, why did you use status 400 instead of 500?

Reply:
Cannot find Announcements เป็นข้อความที่โอเคไหมเพราะเป็น error message ของ get All หรือว่าผิดหลักแกรมม่าภาษาอังกฤษรึป่าวคับ

Reply:
ใน controllers/campground มี error res.status เป็น 400 เหมือนกันคับ

Reply:
แต่มันไม่ควรเป็น 400 ไม่ใช่หรอ ลองอ่านอันนี้ https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400